### PR TITLE
Add documentation for Magellan AI (Actions) Destination

### PR DESCRIPTION
**Pull request by leboshi**
### Proposed changes

* Add documentation for [new action destination](https://github.com/segmentio/action-destinations/pull/2008) to connect Segment with [Magellan AI](https://www.magellan.ai)'s measurement and attribution platform ([more info](https://help.magellan.ai/)).

### Merge timing

Pending approval of the [actual action destination PR](https://github.com/segmentio/action-destinations/pull/2008)

### Notes

Documentation contains a comment at the end about important notes for three action mappings:
* The Magellan AI API spec requires a user agent for the `Install` and `Third-party event` mobile app actions, but the Segment iOS library does not provide this in the event context. Alternatives are suggested.
* There isn't a clear 1:1 between the Segment `Signed Up` B2B SaaS event and our `Lead` action, which is more of an e-commerce action but doesn't directly correlate with any of the Segment spec e-commerce events, so mapping suggestions are provided.

Thank you, and let us know what next steps you need from us!

### CODE REVIEW

File: src/connections/destinations/catalog/actions-magellan-ai/index.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+title: Magellan AI (Actions) Destination
+---
```
Comment (tcgilbert) [inline src/connections/destinations/catalog/actions-magellan-ai/index.md:3]:
```suggestion
---
title: Magellan AI (Actions) Destination
id: 661eca176680eee35d82c955
beta: true
hidden: true
---
```

**Review by tcgilbert — CHANGES_REQUESTED** [review]
in addition to the requested change you will need to change the folder name from `actions-magellan-ai` to `magellan-ai` as this is the slug of your integration

**Comment by leboshi** [comment]
Great!  Moved the slug, and committed the ID metadata.  Thanks!

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+This destination is maintained by Magellan AI. For any issues with the destination, [contact their Measurement team](mailto:measurement@magellan.ai).
+
+## Getting started
+
+1. From your workspace's [Destination catalog page](https://app.segment.com/goto-my-workspace/destinations/catalog){:target="_blank”} search for "Magellan AI"
+2. Select "Magellan AI" and click **Add Destination**
+3. Select an existing Source to connect to Magellan AI (Actions)
+4. Enter the campaign's **pixel token**, either from the [Magellan AI pixel management dashboard](https://app.magellan.ai/navigator/measurement/pixels){:target="_blank"} or provided by your Magellan AI measurement account representative, in the destination settings in Segment
+5. Configure which actions you want to send to Magellan AI
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:20]: hey quick question - is "measurement" *not* capitalized like it is here, or is it capitalized like it is on line 12?

```suggestion
1. From your Segment workspace's [Destination catalog page](https://app.segment.com/goto-my-workspace/destinations/catalog){:target="_blank”} search for "Magellan AI".
2. Select "Magellan AI" and click **Add Destination**.
3. Select an existing Source to connect to Magellan AI (Actions), give your destination a name and click **Create destination**. 
4. Enter the campaign's **pixel token** in the destination settings in Segment. You can obtain this token either from the [Magellan AI pixel management dashboard](https://app.magellan.ai/navigator/measurement/pixels){:target="_blank"} or provided by your Magellan AI Measurement account representative. 
5. Configure which actions you want to send to Magellan AI.
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Optionally, if you need Magellan AI to process GDPR deletion requests:
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:22]:
```suggestion
(Optional) If you need Magellan AI to process GDPR deletion requests:
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+1. Contact your Magellan AI measurement account representative to enable API access for your account
+2. Go to the [Magellan AI API access token page](https://app.magellan.ai/api_access_tokens){:target="_blank"}, and generate a new API token if necessary
+3. Find and copy the new **API token**
+4. Enter the **API token** in the destination settings in Segment
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:26]:
```suggestion
1. Contact your Magellan AI measurement account representative to enable API access for your account.
2. Go to the [Magellan AI API access token page](https://app.magellan.ai/api_access_tokens){:target="_blank"} and generate a new API token.
3. Find and copy the new **API token**.
4. Enter the **API token** in the destination settings for your Magellan AI (Actions) destination.
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+{% include components/actions-fields.html %}
+
+## Limitations
+
+* Magellan AI only supports Segment's Replay feature for mobile events.
+
+[
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:34]:
```suggestion
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+### Lead
+
+Magellan AI's `Lead` action is semantically closest to Segment's B2B SaaS `Signed Up` event and uses it as the default Trigger. However, Magellan AI's API spec considers `Lead` an e-commerce event, requiring a value and a currency. You may:
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:37]:
```suggestion
Magellan AI's `Lead` action is semantically closest to Segment's [B2B SaaS `Signed Up` event](/docs/connections/spec/b2b-saas/#signed-up) and uses it as the default Trigger. However, Magellan AI's API spec considers `Lead` an e-commerce event, requiring a value and a currency. You may:
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+* Configure your sources sending `Signed Up` events to include the additional e-commerce-style fields
+* Consider mapping an alternative event to the `Lead` action, such as `Promotion Clicked` or `Product Added to Wishlist`, depending on your use case
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:39]:
```suggestion
* Consider mapping an alternative event to the `Lead` action, like `Promotion Clicked` or `Product Added to Wishlist`, depending on your use case
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+* Map `Signed Up` events to the `Lead` action, providing dummy values in the mapping like `0` for the value and `USD` for the currency
+
+### Install, Third-Party Event
+
+Magellan AI's API spec requires a user agent, but Segment's iOS library does not provide user agent information in the event context. In order to use this action with the Segment's iOS library, you may provide either a static user agent string or a placeholder value in the mapping.
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:44]:
```suggestion
Magellan AI's API spec requires a user agent, but Segment's [Analytics-Swift](/docs/connections/sources/catalog/libraries/mobile/apple/) library does not provide user agent information in the event context. In order to use this action with Segment's Swift library, you can provide either a static user agent string or a placeholder value in the mapping.
```

File: src/connections/destinations/catalog/magellan-ai/index.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+]: #
```
Comment (forstisabella) [inline src/connections/destinations/catalog/magellan-ai/index.md:45]:
```suggestion
```

**Review by forstisabella — COMMENTED** [review]
a few changes from me and we'll be ready to ship this! 🚀

File: src/connections/destinations/catalog/magellan-ai/index.md (hunk 2 above)
Comment (leboshi) [reply src/connections/destinations/catalog/magellan-ai/index.md:20]: I know this is literally your job, but the absolute glee I feel in someone *caring* about questions like that is so refreshing.  Boss man says big M.  Thank you!

**Comment by leboshi** [comment]
Thank you for the thorough pass! Y'all are incredible. Also updated the actual job title, to appease The People Upstairs when I asked about the capitalization.

**Review by forstisabella — APPROVED** [review]

**Comment by forstisabella** [comment]
sweet!! Thanks so much for giving this PR so much thought. We're good to go, but I'll hold this until the corresponding PR in the action-destination repo is complete!

**Comment by leboshi** [comment]
Hey @forstisabella!  The [destination PR](https://github.com/segmentio/action-destinations/pull/2008) has been approved and deployed.  However, I just noticed that both the reviewer and I missed that our slug is still `actions-magellan-ai` instead of `magellan-ai` in the [metadata for the destination](https://github.com/segmentio/action-destinations/blob/7ea11b45321be4f1d2d12a8fb7223b27e4009ac2/packages/destination-actions/src/destinations/magellan-ai/index.ts#L17), per @tcgilbert's request above.  Soooooo there may already be a fix PR going up, as well, but the destination does indeed exist now!  🙈

**Comment by leboshi** [comment]
OK, the contribution guide says
> Do not change the name or slug of your Integration after the Integration has been deployed.

So I guess we're `actions-magellan-ai` now!  Reverted the previous directory rename to match the slug.

**Comment by forstisabella** [comment]
hi @leboshi - sweet, thanks for the update! I'll merge this docs PR and check after tomorrow's deploy that the link resolves as it should. If it doesn't, I can open a PR for a redirect. thanks for all of your work on these docs!

**Pull request merged**